### PR TITLE
Added support clang 14+ new directory structure

### PR DESCRIPTION
**Pull request by mboisson**

### CODE REVIEW

**Comment by mboisson** [comment]
Don't merge yet, this is not the right fix

**Comment by mboisson** [comment]
:( apparently, this never got merged
https://github.com/easybuilders/easybuild-easyblocks/pull/2264
so this PR can't be merged as is...

**Comment by mboisson** [comment]
In fact, I am not able to get clang to build without downloading the full package `llvm-project-%(version)s.src.tar.xz`. The sum of components does not equal the whole thing, there are parts missing. 

I would suggest to drop the per-component download, and to select which component to install based on `LLVM_ENABLE_PROJECTS`

File: easybuild/easyblocks/c/clang.py
Comment (lexming) [inline easybuild/easyblocks/c/clang.py:122]: It is confusing that flags can overwrite `llvm_projects` in version >=14 but is the opposite for versions <14. Can we consolidate these settings in a common variable and use that single source of settings in the rest of the easyblock? For instance, I think we can always use `llvm_projects` and replace checks for versions <14 such as `if self.cfg["usepolly"]` for `if "polly" in llvm_projects`.

File: easybuild/easyblocks/c/clang.py
Comment (lexming) [inline easybuild/easyblocks/c/clang.py:181]: The tarballs `llvm-project-14.0.0.src` and `llvm-project-13.0.0.src` don't look that different and LLVM v13 supports `LLVM_ENABLE_PROJECTS` as well.

Instead of checking for version 14+ or not, should we check if the sources of the easyconfig have an `llvm-project-*` tarball, and if so proceed in the new way with `LLVM_ENABLE_PROJECTS`?

File: easybuild/easyblocks/c/clang.py
Comment (mboisson) [reply easybuild/easyblocks/c/clang.py:122]: I wanted to minimize changes, but sure, I can do that. For some however, the condition is trickier.

File: easybuild/easyblocks/c/clang.py
Comment (mboisson) [reply easybuild/easyblocks/c/clang.py:181]: sure, I can do that as well

File: easybuild/easyblocks/c/clang.py
Comment (mboisson) [reply easybuild/easyblocks/c/clang.py:122]: addressed in ae86d8876

File: easybuild/easyblocks/c/clang.py
Comment (mboisson) [reply easybuild/easyblocks/c/clang.py]: addressed in ae86d8876

**Comment by ocaisa** [comment]
@mboisson This looks good, but there seems to be a conflict right now, can you fix that?

Is there a PR that uses this? I can add one based on your repos. 

I'd like to do one which includes `flang` from v15 up. Can we add that to the sanity check step?
```
        if 'flang' in self.cfg['llvm_projects'] and LooseVersion(self.version) >= LooseVersion('15'):
            custom_paths['files'].extend(["bin/flang"])
```

File: easybuild/easyblocks/c/clang.py
Comment (ocaisa) [inline easybuild/easyblocks/c/clang.py:103]: Should we curate this list and have accepted values? With this PR we would support
```
llvm_projects = ['polly', 'lld', 'libunwind', 'lldb', 'libcxx', 'libcxxabi', 'clang-tools-extra']
```
but what if someone goes outside this group? We could let them do it but throw a warning that the project is not currently supported by the easyblock but that we will include it in the build anyway.

File: easybuild/easyblocks/c/clang.py
Comment (ocaisa) [inline easybuild/easyblocks/c/clang.py:120]: This could be the place to check
```
            self.cfg['llvm_projects'] = []
        else:
            for project in self.cfg['llvm_projects']:
                if project not in KNOWN_LLVM_PROJECTS:
                    warn_msg = "LLVM project %s included but not recognised, this project will NOT be sanity checked!" % project
                    self.log.warning(warn_msg)
                    print_warning(warn_msg)
```

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)

Build succeeded for 0 out of 0 (3 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/8a509b460741d79ab8c27a9f5b35d352 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-13.0.1-GCCcore-11.2.0.eb_ (partial log available at https://gist.github.com/ce472517e5a4155d369e8b728d7875bf)
 * **FAIL (build issue)** _Clang-11.0.1-GCCcore-10.2.0.eb_ (partial log available at https://gist.github.com/d36e6cc98b1380816af08feee03018c0)
 * **FAIL (build issue)** _Clang-10.0.1-GCCcore-9.3.0.eb_ (partial log available at https://gist.github.com/80cf177efddf9baa073617abc2457f99)

Build succeeded for 0 out of 3 (3 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/6c6a29fb4586ce4b988cebaa081048be for a full test report.

**Comment by mboisson** [comment]
> @mboisson This looks good, but there seems to be a conflict right now, can you fix that?
I have no clue how to merge that conflict... the code is completely different and is not in a single commit :(

**Comment by mboisson** [comment]
I tried to resolve the conflicts by cherry-picking all missing commits to this file and manually fixing the conflict when cherry-picking the commit that caused it, but Github still does not like it. A maintainer will have to fix it when merging. @ocaisa

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-13.0.1-GCCcore-11.2.0.eb_ (partial log available at https://gist.github.com/87ce834ed7ac4be337eef83529b18b0b)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/1319a4d0b08eb69b3c2329835a6e122d for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/85fc28ed1e4a6e4eb273f5e3e9e93ec9 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-11.0.1-GCCcore-10.2.0.eb_ (partial log available at https://gist.github.com/63ee9dc613f8649e22750d30d43ea680)
 * **SUCCESS** _Clang-12.0.1-GCCcore-10.3.0.eb_ 

Build succeeded for 1 out of 2 (2 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/9e932dfd1536a802a52dd5cb4e6f1ca4 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-15.0.4-GCCcore-11.3.0.eb_ (partial log available at https://gist.github.com/fe7624019f4b83e7a4c1455defc86e54)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
cnx2 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/ca9aa52c18f12124cd60c2d11d414dd0 for a full test report.

File: easybuild/easyblocks/c/clang.py
Comment (ocaisa) [inline easybuild/easyblocks/c/clang.py:635]:
```suggestion
            custom_paths['files'].extend(["bin/flang-new"])
```
Sorry, my mistake, right now that compiler is called `flang-new`

Comment (ocaisa) [reply easybuild/easyblocks/c/clang.py:635]: It will change at some point to `flang` (but I can't find a timeline)

Comment (ocaisa) [reply easybuild/easyblocks/c/clang.py:635]: `flang-new` is the one recognised by CMake 3.24.0+

**Comment by ocaisa** [comment]
The failing test for Clang 11 is related to the sanity check, there is no `lib/libomptarget.rtl.x86_64.so`. That's not related to this PR though, so no need to fix here.

The failure for Clang 15 is partially related to the fact the compiler is called `flang-new` but also to
```
no file found at 'lib/libc++.so' in /home/ocaisa/.local/easybuild/software/Clang/15.0.4-GCCcore-11.3.0
no file found at 'lib/libc++abi.so' in /home/ocaisa/.local/easybuild/software/Clang/15.0.4-GCCcore-11.3.0
```

Looking at the configure step I see for Clang 15 I see:
```
CMake Warning at CMakeLists.txt:101 (message):
  Using LLVM_ENABLE_PROJECTS=libcxx is deprecated now, please use
  -DLLVM_ENABLE_RUNTIMES=libcxx or see the instructions at
  https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.


CMake Warning at CMakeLists.txt:101 (message):
  Using LLVM_ENABLE_PROJECTS=libcxxabi is deprecated now, please use
  -DLLVM_ENABLE_RUNTIMES=libcxxabi or see the instructions at
  https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.


CMake Warning at CMakeLists.txt:101 (message):
  Using LLVM_ENABLE_PROJECTS=libunwind is deprecated now, please use
  -DLLVM_ENABLE_RUNTIMES=libunwind or see the instructions at
  https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.

```

**Comment by mboisson** [comment]
> The failing test for Clang 11 is related to the sanity check, there is no `lib/libomptarget.rtl.x86_64.so`. That's not related to this PR though, so no need to fix here.
> 
> The failure for Clang 15 is partially related to the fact the compiler is called `flang-new` but also to
> 
> ```
> no file found at 'lib/libc++.so' in /home/ocaisa/.local/easybuild/software/Clang/15.0.4-GCCcore-11.3.0
> no file found at 'lib/libc++abi.so' in /home/ocaisa/.local/easybuild/software/Clang/15.0.4-GCCcore-11.3.0
> ```
> 
> Looking at the configure step I see for Clang 15 I see:
> 
> ```
> CMake Warning at CMakeLists.txt:101 (message):
>   Using LLVM_ENABLE_PROJECTS=libcxx is deprecated now, please use
>   -DLLVM_ENABLE_RUNTIMES=libcxx or see the instructions at
>   https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.
> 
> 
> CMake Warning at CMakeLists.txt:101 (message):
>   Using LLVM_ENABLE_PROJECTS=libcxxabi is deprecated now, please use
>   -DLLVM_ENABLE_RUNTIMES=libcxxabi or see the instructions at
>   https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.
> 
> 
> CMake Warning at CMakeLists.txt:101 (message):
>   Using LLVM_ENABLE_PROJECTS=libunwind is deprecated now, please use
>   -DLLVM_ENABLE_RUNTIMES=libunwind or see the instructions at
>   https://libcxx.llvm.org/BuildingLibcxx.html for building the runtimes.
> ```

when did that become deprecated ?! I built 15.0.2 without issue

**Comment by mboisson** [comment]
🤔 I do have these warnings as well, but somehow the sanity checks did not fail for my build, even though those libraries are indeed missing. I'll need to investigate (probably next week)

**Comment by mboisson** [comment]
🤔 
```
== 2022-10-14 06:28:44,906 build_log.py:266 INFO sanity checking...
== 2022-10-14 06:28:44,908 easyblock.py:3820 INFO Starting sanitycheck step
== 2022-10-14 06:28:44,909 easyblock.py:3828 INFO Running method sanity_check_step part of step sanitycheck
== 2022-10-14 06:28:44,909 easyblock.py:3007 INFO Running sanity check with extra modules: python/3.8
== 2022-10-14 06:28:44,910 easyblock.py:3243 INFO Using (only) sanity check paths specified by easyconfig file: {'dirs': ['lib/python3.8/site-packages'], 'files': []}
== 2022-10-14 06:28:44,910 easyblock.py:3291 INFO Using customised sanity check commands: ['clang --help', 'clang++ --help', 'llvm-config --cxxflags', "python -c 'import clang'"]
== 2022-10-14 06:29:08,831 easyblock.py:3484 INFO Loading extra modules for sanity check: python/3.8
== 2022-10-14 06:29:09,251 easyblock.py:3506 INFO sanity check command clang --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:09,285 easyblock.py:3506 INFO sanity check command clang++ --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:09,296 easyblock.py:3506 INFO sanity check command llvm-config --cxxflags ran successfully! (output: -I/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/clang/15.0.2/include -std=c++14    -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
== 2022-10-14 06:29:10,026 easyblock.py:3506 INFO sanity check command python -c 'import clang' ran successfully! (output: )
== 2022-10-14 06:29:11,021 easyblock.py:3007 INFO Running sanity check with extra modules: python/3.9
== 2022-10-14 06:29:11,021 easyblock.py:3243 INFO Using (only) sanity check paths specified by easyconfig file: {'dirs': ['lib/python3.9/site-packages'], 'files': []}
== 2022-10-14 06:29:11,021 easyblock.py:3291 INFO Using customised sanity check commands: ['clang --help', 'clang++ --help', 'llvm-config --cxxflags', "python -c 'import clang'"]
== 2022-10-14 06:29:19,531 easyblock.py:3484 INFO Loading extra modules for sanity check: python/3.9
== 2022-10-14 06:29:19,569 easyblock.py:3506 INFO sanity check command clang --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:19,602 easyblock.py:3506 INFO sanity check command clang++ --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:19,613 easyblock.py:3506 INFO sanity check command llvm-config --cxxflags ran successfully! (output: -I/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/clang/15.0.2/include -std=c++14    -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
== 2022-10-14 06:29:20,835 easyblock.py:3506 INFO sanity check command python -c 'import clang' ran successfully! (output: )
== 2022-10-14 06:29:21,766 easyblock.py:3007 INFO Running sanity check with extra modules: python/3.10
== 2022-10-14 06:29:21,766 easyblock.py:3243 INFO Using (only) sanity check paths specified by easyconfig file: {'dirs': ['lib/python3.10/site-packages'], 'files': []}
== 2022-10-14 06:29:21,766 easyblock.py:3291 INFO Using customised sanity check commands: ['clang --help', 'clang++ --help', 'llvm-config --cxxflags', "python -c 'import clang'"]
== 2022-10-14 06:29:30,818 easyblock.py:3484 INFO Loading extra modules for sanity check: python/3.10
== 2022-10-14 06:29:30,852 easyblock.py:3506 INFO sanity check command clang --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:30,884 easyblock.py:3506 INFO sanity check command clang++ --help ran successfully! (output: OVERVIEW: clang LLVM compiler
== 2022-10-14 06:29:30,894 easyblock.py:3506 INFO sanity check command llvm-config --cxxflags ran successfully! (output: -I/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/clang/15.0.2/include -std=c++14    -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
== 2022-10-14 06:29:30,923 easyblock.py:3506 INFO sanity check command python -c 'import clang' ran successfully! (output: )
``` 

Is it possible that having extensions somehow skips the main package sanity checks ?

**Comment by mboisson** [comment]
Ah, I guess we should set `enhance_sanity_check` in our hooks

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-15.0.5-GCCcore-11.3.0.eb_ (partial log available at https://gist.github.com/7a1095e350d35c7b99c66de94df4e8a0)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/808ddb50353e531bd7ec143ef5a8d970 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0.eb_ 
 * **FAIL (build issue)** _Clang-12.0.1-GCCcore-10.3.0.eb_ (partial log available at https://gist.github.com/51b7510dabcf53b229dbce88fc4b44dc)

Build succeeded for 1 out of 2 (2 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/3f835679da1336444009119ae8ed830e for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0.eb_ 
 * **FAIL (build issue)** _Clang-12.0.1-GCCcore-10.3.0.eb_ (partial log available at https://gist.github.com/002ed868816dc48fab4aa67922bd143f)

Build succeeded for 1 out of 2 (2 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/8074b3ddd79ba75b062c6afefb40a811 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/af457d88f14a98778665d5198d4c60dd for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-12.0.1-GCCcore-10.3.0.eb_ (partial log available at https://gist.github.com/c0eb2a369d67388e374363f8efbcb756)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/0f143bb72ce6a4bea893764e70e8ce19 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-12.0.1-GCCcore-10.3.0.eb_ (partial log available at https://gist.github.com/41bf895d4058c33d0875403d8291b253)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/8fb0d2ca0b4e2ab68a355a0ce17ccb82 for a full test report.

File: easybuild/easyblocks/c/clang.py
Comment (ocaisa) [inline easybuild/easyblocks/c/clang.py:82]: So the full list of runtimes is actually longer, see https://llvm.org/docs/GettingStarted.html#local-llvm-configuration and this is "Only effective when using the full monorepo layout" (which I'm guessing means the single source tarball). I'll make a PR to your easyblock to support this.

Comment (ocaisa) [reply easybuild/easyblocks/c/clang.py:82]: PR opened at https://github.com/ComputeCanada/easybuild-easyblocks/pull/15

Comment (mboisson) [reply easybuild/easyblocks/c/clang.py:82]: I saw this, but I noticed that only `libunwind`, `libcxx` and `libcxxabi` end up in the runtime directory, so I assumed this was not yet transitioned.

Comment (mboisson) [reply easybuild/easyblocks/c/clang.py:82]: i.e.: 
```
ls /home/mboisson/.local/easybuild/software/2020/avx2/Core/clang/15.0.2/lib/x86_64-unknown-linux-gnu/
libc++.a  libc++abi.a  libc++abi.so  libc++abi.so.1  libc++abi.so.1.0  libc++experimental.a  libc++.so  libc++.so.1  libc++.so.1.0  libunwind.a  libunwind.so  libunwind.so.1  libunwind.so.1.0
```

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-13.0.1-GCCcore-11.2.0-CUDA-11.4.1.eb_ (partial log available at https://gist.github.com/cc2cdbbe4a3678d673f2b15d8fcdeeb7)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5 (Green Obsidian), x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.10.4
See https://gist.github.com/45c75ac6b34363cc4389b1122ccc59d7 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-12.0.1-GCCcore-10.3.0.eb_ (partial log available at https://gist.github.com/9b48eede6c3a6d5124e2bb24c9de3560)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
cnx3 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/2b1158a9e05d3536fb88db0cfaa323c1 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-13.0.1-GCCcore-11.2.0-CUDA-11.4.1.eb_ (partial log available at https://gist.github.com/e215a491922334980da9423f18edfcaf)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/7591df2b80ede480394319aff3feb7ef for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0-CUDA-11.4.1.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/30074b89aa9e6d4a981425783a822b46 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-12.0.1-GCCcore-10.3.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/f975e73b47250690c6927ef0fdd3fed6 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0-CUDA-11.4.1.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
cnx2 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/c721ea8c70087972f3f580efd3f048ea for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0-CUDA-11.4.1.eb_ 
 * **SUCCESS** _Clang-11.0.1-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _Clang-12.0.1-GCCcore-10.3.0.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/e860f312fedad6fe181042367bc359dc for a full test report.

**Comment by ocaisa** [comment]
Last test was to ensure `--module-only` was also now working for Clang

**Review by ocaisa — APPROVED** [review]
Thanks for all the hard work @mboisson !

**Comment by mboisson** [comment]
> Thanks for all the hard work @mboisson !

🥳

File: easybuild/easyblocks/c/clang.py
Comment (boegel) [inline easybuild/easyblocks/c/clang.py:184]: Doing this in the constructor doesn't work, since the modules for the dependencies are only loading in `prepare_step` => https://github.com/easybuilders/easybuild-easyblocks/issues/2871

Comment (ocaisa) [reply easybuild/easyblocks/c/clang.py:184]: I guess the key is in the comment here, this should be like what is done in LAMMPS (https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/l/lammps.py#L175-L177)

Comment (boegel) [reply easybuild/easyblocks/c/clang.py:184]: @ocaisa please mention that in #2871

**Pull request merged**
